app/partials/data-export-button.cjsx

```diff
@@ -14,15 +14,37 @@ module.exports = React.createClass
   getInitialState: ->
     exportRequested: false
     exportError: null
+    showButton: true
+    mostRecent: null
+
+  componentDidMount: ->
+    @exportGet()
+    @showDataExportBtn()
+
+  componentWillReceiveProps: ->
```

Comment:
I don't think defining `componentWillReceiveProps` is necessary since none of the props defined are expected to change while the component is in use (project will only change if the user navigates away to their project list or directly modify the URL anyway at which point the component will mount the first time and call `componentDidMount`)

As an aside, if there is a prop that is expected to change, you typically would want to add a conditional to determine whether or not to call other methods in `componentWillReceiveProps` since it may be a different prop that is changing (let's say the boolean prop in this case if it was changing based on a parent component's state change and if it was written that way). For example (note I'm writing this in coffeescript since this component is still written using it):
```
componentWillReceiveProps: (nextProps) ->
  if @props.project isnt nextProps.project
    @doSomething()
```
Without the conditional, the method will always be called regardless of which prop is changing.